com.unity.render-pipelines.high-definition/Editor/Material/Decal/DecalProjectorEditor.cs

```diff
@@ -150,22 +150,15 @@ public Bounds OnGetFrameBounds()
         public void UpdateMaterialEditor()
         {
             int validMaterialsCount = 0;
-            for (int index = 0; index < targets.Length; ++index)
```

Comment:
So this problematic loop on target. 

It has no meanings on this file l.112 as it is used as a hook on each DecalProjector when the material change. So there we should not use the target notion. We should have a DecalProjector parameter to gather changes.

On line 109 though, the same function is called from OnEnable where multiselection can work on several target at a time.

It seams that this function (original version) is OK for line 109 but we should add another dedicated callback for the individual updated used in event

Reply:
The problem seems to come from another direction.